lib/facter/extlib__puppet_config.rb

```diff
@@ -31,6 +31,10 @@
       end
     end
 
+    # confdir can't be defined in the config file
+    # because puppet uses it to find the config file
+    puppet_config['confdir'] = Puppet.settings['confdir']
```

Comment:
Wouldn't it be easier to add it around line 5?

Reply:
I can move the block further up, but I didn't want to break up the existing logic for how the settings are imported.

Would something like this work for you:
```ruby
Facter.add(:extlib__puppet_config) do
  setcode do
    puppet_config = {}

    # confdir can't be defined in the config file
    # because puppet uses it to find the config file
    puppet_config['confdir'] = Puppet.settings['confdir']

    desired_settings = {
<snip>
```

Reply:
I was hinting at defining the hash with the value initialized instead of adding it after the fact.
```ruby
puppet_config = {
  'confdir' => Puppet.settings['confdir'],
}
```